src/core/algorithms/fd/fdep/fd_tree_element.cpp

```diff
@@ -243,15 +243,16 @@ void FDTreeElement::PrintDependencies(std::bitset<kMaxAttrNum>& active_path, std
     }
 }
 
-void FDTreeElement::FillFdCollection(RelationalSchema const& scheme,
-                                     std::list<FD>& fd_collection) const {
+void FDTreeElement::FillFdCollection(RelationalSchema const& scheme, std::list<FD>& fd_collection,
+                                     unsigned int max_lhs_ = -1) const {
```

Comment:
```suggestion
                                     unsigned int max_lhs_ = std::numeric_limits<unsigned int>::max()) const {
```

Reply:
And naming is incorrect btw, `_` postfix is redundant

Reply:
Also, `max_lhs` parameter has default value here, but it's seen only in the definition of this class, which is rather useless, because this method is never called internally.
So either place the default parameter to the declaration or remove it :)